app/scripts/components/common/mapbox/layers/vector-timeseries.tsx

```diff
@@ -0,0 +1,242 @@
+      } catch (error) {
+        if (!controller.signal.aborted) {
+          setFeaturesApiEndpoint('');
+          onStatusChange?.({ status: S_FAILED, id });
+        }
+        return;
+      }
+    }
+
+    load();
+
+    return () => {
+      controller.abort();
+    };
+  }, [mapInstance, id, stacCol, date, onStatusChange]);
+
+  //
+  // Generate Mapbox GL layers and sources for raster timeseries
+  //
+  const haveSourceParamsChanged = useMemo(
+    () => JSON.stringify(sourceParams),
+    [sourceParams]
+  );
+  useEffect(() => {
+    if (!date || !featuresApiEndpoint) return;
+
+    const start = userTzDate2utcString(startOfDay(date));
+    const end = userTzDate2utcString(endOfDay(date));
+
+    const tileParams = qs.stringify({
+      ...sourceParams,
+      datetime: `${start}/${end}`
+    });
+
+    const sources: Record<string, AnySourceImpl> = {
+      [id]: {
+        type: 'vector',
+        tiles: [`${featuresApiEndpoint}/tiles/{z}/{x}/{y}?${tileParams}`]
+      } as VectorSourceImpl
+    };
+
+    const layers = [
+      {
+        id: `${id}-line-bg`,
+        type: 'line',
+        source: id,
+        'source-layer': 'default',
+        layout: {
+          visibility: isHidden ? 'none' : 'visible'
+        },
+        paint: {
+          'line-color': theme.color?.['base-300'],
+          'line-width': 4
+        },
+        // filter: ['==', '$type', 'LineString'],
+        minzoom: minZoom,
+        metadata: {
+          layerOrderPosition: 'raster'
+        }
+      },
+      {
+        id: `${id}-line-fg`,
+        type: 'line',
+        source: id,
+        'source-layer': 'default',
+        layout: {
+          visibility: isHidden ? 'none' : 'visible'
+        },
+        paint: {
+          'line-color': theme.color?.primary,
+          'line-width': 2
+        },
+        filter: ['==', '$type', 'LineString'],
+        minzoom: minZoom,
+        metadata: {
+          layerOrderPosition: 'raster'
+        }
+      },
+      {
+        id: `${id}-fill-fg`,
+        type: 'fill',
+        source: id,
+        'source-layer': 'default',
+        layout: {
+          visibility: isHidden ? 'none' : 'visible'
+        },
+        paint: {
+          'fill-color': theme.color?.primary,
+          'fill-opacity': 0.8
+        },
+        filter: ['==', '$type', 'Polygon'],
+        minzoom: minZoom,
+        metadata: {
+          layerOrderPosition: 'raster'
+        }
+      },
+      minZoom > 0
+        ? {
+            type: 'symbol',
+            id: `${id}-points`,
+            source: id,
+            'source-layer': 'default',
+            layout: {
+              'icon-image': 'leaflet-marker',
+              visibility: isHidden ? 'none' : 'visible',
+              'icon-offset': [0, -12]
+            },
+            maxzoom: minZoom,
+            metadata: {
+              layerOrderPosition: 'markers'
+            }
+          }
+        : undefined
+    ].filter(Boolean) as AnyLayer[];
```

Comment:
Interesting! In this case we're not mapping anything, but constructing an array.
Maybe we could do:
```
const values = [1, 2, 3, condition ? 4 : []].flat()
```
Was this what you meant?

In this case I think this would be harder to read.